### PR TITLE
Adds CPMap Client Support [HZ-3539]

### PR DESCRIPTION
**Pull request by gbarnett-hz**
Adds client support for the CPMap API, e.g.

```java
HazelcastInstance client = HazelcastClient.newHazelcastClient();
CPMap<String, String> map = client.getCPSubsystem().getMap("mymap");
assert map.put("k", "v") == null;
assert "v".equals(map.get("k"));
```

EE PR: https://github.com/hazelcast/hazelcast-enterprise/pull/6752
Client Protocol PR: https://github.com/hazelcast/hazelcast-client-protocol/pull/484

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Add `Add to Release Notes` label if changes should be mentioned in release notes or `Not Release Notes content` if changes are not relevant for release notes
- [ ] Request reviewers if possible
- [ ] New public APIs have `@Nonnull/@Nullable` annotations
- [ ] New public APIs have `@since` tags in Javadoc
- [ ] Send backports/forwardports if fix needs to be applied to past/future releases

### CODE REVIEW

**Review by pveentjer — APPROVED** [review]

**Comment by gbarnett-hz** [comment]
@hazelcast/apis could you take a look at this please?

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/DefaultMessageTaskFactoryProvider.java [hunk 1]
```diff
@@ -873,9 +873,9 @@
 public class DefaultMessageTaskFactoryProvider implements MessageTaskFactoryProvider {
     private static final int MESSAGE_TASK_PROVIDER_INITIAL_CAPACITY = 500;
 
-    private final Int2ObjectHashMap<MessageTaskFactory> factories;
+    protected final Int2ObjectHashMap<MessageTaskFactory> factories;
```
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/client/impl/protocol/DefaultMessageTaskFactoryProvider.java:876]: nit: since this class is no longer extended on EE side, can revert these changes

**Review by vbekiaris — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/DefaultMessageTaskFactoryProvider.java (hunk 1 above)
Comment (gbarnett-hz) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/DefaultMessageTaskFactoryProvider.java:876]: Agree, will revert tomorrow.

Comment (gbarnett-hz) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/DefaultMessageTaskFactoryProvider.java]: Done for both fields.

**Review by srknzl — APPROVED** [review]
